core/src/main/java/org/apache/calcite/rex/RexAnalyzer.java

```diff
@@ -128,6 +128,10 @@ public RexAnalyzer(RexNode e, RelOptPredicateList predicates) {
     }
 
     @Override public Void visitCall(RexCall call) {
+      if (!call.getOperator().isDeterministic()) {
+        ++unsupportedCount;
```

Comment:
Can you please clarify why do you think `RexInterpreter` can't handle non-deterministic functions?

Reply:
It throws exception without these codes.

Reply:
Does it? Can you please clarify what is the exception?

Reply:
Actually RexInterpreter supports quite a few functions[1]. 
```
java.lang.IllegalArgumentException: unbound: NDC()
	at org.apache.calcite.rex.RexInterpreter.unbound(RexInterpreter.java:71)
	at org.apache.calcite.rex.RexInterpreter.visitCall(RexInterpreter.java:203)
	at org.apache.calcite.rex.RexInterpreter.visitCall(RexInterpreter.java:47)
	at org.apache.calcite.rex.RexCall.accept(RexCall.java:288)
	at org.apache.calcite.rex.RexInterpreter.visitCall(RexInterpreter.java:129)
	at org.apache.calcite.rex.RexInterpreter.visitCall(RexInterpreter.java:47)
	at org.apache.calcite.rex.RexCall.accept(RexCall.java:288)
	at org.apache.calcite.rex.RexInterpreter.evaluate(RexInterpreter.java:63)
	at org.apache.calcite.rex.RexSimplify.verify(RexSimplify.java:1796)
	at org.apache.calcite.rex.RexSimplify.simplifyUnknownAs(RexSimplify.java:233)
	at org.apache.calcite.rex.RexProgramTestBase.checkSimplify3_(RexProgramTestBase.java:127)
	at org.apache.calcite.rex.RexProgramTestBase.checkSimplifyUnchanged(RexProgramTestBase.java:92)
	at org.apache.calcite.rex.RexProgramTest.testSimplifyNonDeterministicFunction(RexProgramTest.java:2569)
```

[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rex/RexInterpreter.java#L126-L204

Reply:
If it can't handle lots of things that is fine. It is not fine to exclude `isDeterministic` here because one day `rand()` function might get supported in `RexAnalyzer`, and it would be hard to tell why it is not used.

How about adding a method like `RexInterpreter#isSupported(RexCall)`?

Reply:
It might be better to put ```isSupported(RexCall)`` into ```VariableCollector``` which is used to counts features (such as CAST) that ```RexInterpreter``` cannot handle[1].

[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rex/RexAnalyzer.java#L105-L109.